shared/src/main/scala/coop/rchain/store/KeyValueTypedStore.scala

```diff
@@ -9,5 +9,12 @@ trait KeyValueTypedStore[F[_], K, V] {
 
   def contains(keys: Seq[K]): F[Seq[Boolean]]
 
+  /**
+    * Efficient way to iterate and filter the whole KV store
+    *
+    * @param pf Partial function to project and filter values
+    */
+  def collect[T](pf: PartialFunction[(K, () => V), T]): F[Seq[T]]
```

Comment:
It's also easy to convert function returning `Option` to `PartialFunction`.
https://www.scala-lang.org/files/archive/api/2.13.x/scala/Function$.html#unlift[T,R](f:T=%3EOption[R]):PartialFunction[T,R]